packages/kas/src/nodes.ts

```diff
@@ -3051,18 +3121,22 @@ export class Const extends Symbol {
 }
 
 /* abstract number node */
-class Num extends Expr {
+abstract class Num extends Expr {
+    n: number;
+
     constructor() {
         super();
+        this.n = 0;
```

Comment:
So this is just a placeholder because Num needs `n` for `getDecimalPlaces`? But really `n` gets set by its child classes? Just making sure I understand the motivation.

Also just curious, could this also be `n: number = 0;`?

Reply:
This is to make TypeScript happy.  We could do `n: number = 0;`, but I tend to prefer setting variables in the constructor when there is a constructor.  I find it hard to reason about the order things run when we're doing both.  In this case I don't see any harm in having `n: number = 0;` outside the constructor.